op-service/sources/receipts_basic.go

```diff
@@ -54,6 +55,22 @@ func (f *BasicRPCReceiptsFetcher) FetchReceipts(ctx context.Context, blockInfo e
 	return res, nil
 }
 
+// BatchFetchReceipts is inefficient for this implementation, it just calls FetchReceipts for each blockInfo.
+func (f *BasicRPCReceiptsFetcher) BatchFetchReceipts(ctx context.Context, blockInfos []eth.BlockInfo, txHashes [][]common.Hash) ([]types.Receipts, error) {
```

Comment:
I'm torn on trying to make the Basic Fetcher do better Batch calls. It would be a benefit, but if a client is using this method, they're already missing out on the performance of `eth_GetBlockReceipts`. Will make this an efficient function if there is demand for it.

Reply:
I would like to leave the basic implementation basic and therefore don't plan on implementing batch-unioning in the basic receipt fetcher. if anyone is strongly against that idea LMK :)